rdkit/ML/Scoring/UnitTestScoring.py

```diff
@@ -147,6 +147,38 @@ def test4_BEDROC(self):
     bedroc = Scoring.CalcBEDROC(self.scoreAllDecoys, self.index, self.alpha)
     self.assertEqual(bedroc, 0.0)
 
+  def test_safeguard_sort(self):
+
+    # Garbage scores with sorted classes show garbage performance
```

Comment:
You may feel like this is garbage, but it is just what it is, an incorrect estimate of ROC due to a bias in the data.